webui/react/src/e2e/models/hew/Card.ts

```diff
@@ -10,9 +12,20 @@ import { WorkspaceActionDropdown } from 'e2e/models/components/WorkspaceActionDr
  */
 export class Card extends NamedComponent {
   override defaultSelector: string = ''; // must be provided
+  static ActionMenuSelector = '[aria-label="Action menu"]';
 
-  readonly actionMenu = new WorkspaceActionDropdown({
+  readonly actionMenu: Dropdown = new WorkspaceActionDropdown({
     parent: this,
-    selector: '[aria-label="Action menu"]',
+    selector: Card.ActionMenuSelector,
   });
+
+  static withName<T extends Card>(
```

Comment:
cool! is this like a second constructor? could we use this syntax, or would that not be the same?
```
export class CardWithName<T extends Card>
```

Reply:
Yeah, it's similar to a constructor. I think this factory pattern is the cleanest though. I'm open too ideas but I tried a couple options. 

I did try making this a second constructor on `Card` but this doesn't work because it always returns a `Card`. I don't think `new Card(name, type)` can actually return a `ProjectCard` or `WorkspaceCard`. Maybe it could satisfy the interface based on class shape, but we'd always have to cast it to a more specific type when we work with it which is a pain. If there is clean way to do it I'd prefer this route, but I don't see a good way. 

Likewise, I think a `new CardWithName<ProjectCard>(name)` would return a `CardWithName<ProjectCard>` class. We could then retrieve an instance of `ProjectCard` from it but then we are back to a factory pattern with more steps.

I can't make this a constructor on `ProjectCard` because then it doesn't encapsulate the selector logic so I'd have to re-define that logic for each card type.

Reply:
okay nice. thanks for checking all that! and i think this solution looks clean too. you're doing something pretty different from what i did with tables and rows too.

Reply:
> I can't make this a constructor on ProjectCard because then it doesn't encapsulate the selector logic so I'd have to re-define that logic for each card type.
```
export class CardWithName<T extends Card> extends Card
```
looks a little ridiculous lol